viewing_party/party.py

```diff
@@ -0,0 +1,199 @@
+        for friend_titles in data["watched"]:
+            friends_movie_titles.append(friend_titles["title"])
+
+    unique_titles =[]
+    for title in user_watched_movies:
+        if title["title"] not in friends_movie_titles:
+            unique_titles.append(title) 
+    return unique_titles
+
+# Next three tests are for this function:
+def get_friends_unique_watched(user_data):
+    """
+    This function compares a list of watched movies from user and user's friends.
+    The function returns movies that the at least one friend of the user has watched,
+    but the user has not watched.  
+    """
+    user_watched_movies = user_data["watched"]
+    friends_watched_movies = user_data["friends"]
+    
+    user_titles =[]       
+    for title in user_watched_movies:
+        user_titles.append(title) 
+    
+    unique_friends_movie_titles = []
+    for data in friends_watched_movies:
+        for friend_titles in data["watched"]:
+            if friend_titles not in user_titles:
+                if friend_titles not in unique_friends_movie_titles:
+                    unique_friends_movie_titles.append(friend_titles)
+    return unique_friends_movie_titles
+
+    # WAVE 4
+    # Next 4 tests are about this function:
+def get_available_recs(user_data):
+    """
+    This function compares watched movies between user and user's friends.
+    If the user has not watched the movie and have a subscription to the host, this function will
+    return recommended movies along with the host/subscription. 
+    """
+    recommended_movies = [] 
+    friends = user_data["friends"]
+    
+    # Collectd a list of titles the user has already watched to 
+    # compare to friends watched later.
+    user_watched = []
+    for movie in user_data["watched"]:
+        user_watched.append(movie["title"])
+        
+    for friend in friends:
+        for movie in friend["watched"]:
+            if movie["title"] not in user_watched and movie["host"] in user_data["subscriptions"]:
+                recommended_movies.append(movie)
+                
+    
+    return recommended_movies
+
+# WAVE 5
+# Four tests for this function:
+def get_new_rec_by_genre(user_data):
+    """
+    This function will return a list  of recommendations based off 
+    the users most watched genres and if they have not seen the movie yet. 
+    """
+    recommended_movies = []
+
+    user_watched = user_data["watched"]
+    friends_watched = user_data["friends"]
+    
+    # list of dictionary title  & genre user has watched
+    movies_watched_by_user = []
+    for title in user_watched:
+        movies_watched_by_user.append(title)
+    
+    # list of movies friends have watched
+    friends_watched_movies = []
+    for friend in friends_watched:
+        for movie in friend["watched"]:
+            if movie not in movies_watched_by_user:
+                if movie not in friends_watched_movies:
+                    if movie["genre"] in get_most_watched_genre(user_data):
```

Comment:
I'd recommend condensing these into a single compound conditional to avoid so much indentation:

```suggestion
            if movie not in movies_watched_by_user \
               and movie not in friends_watched_movies \
               and get_most_watched_genre(user_data) # prevent TypeError \
               and movie["genre"] in get_most_watched_genre(user_data) \
               and movie not in recommended_movies:
```

This reads more clearly if you're used to reading Python code.

You also have a bug if `get_most_watched_genre` returns `None`.

This is where it makes sense to see if the movie is already in `recommended_movies`.

Reply:
And, yeah.  These conditions are pretty ugly so breaking things out into different passes of the loop might read more cleanly.